tests/rptest/services/kgo_repeater_service.py

```diff
@@ -205,13 +206,14 @@ def stop(self, *args, **kwargs):
         super().stop(*args, **kwargs)
 
     def stop_node(self, node):
-        if self._pid is None:
+        if node.name in self._pid_by_node:
             return
+        pid = self._pid_by_node[node.name]
 
         self.redpanda.logger.info(f"{self.__class__.__name__}.stop")
-        self.logger.debug("Killing pid %s" % {self._pid})
+        self.logger.debug("Killing pid %s" % {pid})
         try:
-            node.account.signal(self._pid, signal.SIGKILL, allow_fail=False)
+            node.account.signal(pid, signal.SIGKILL, allow_fail=False)
```

Comment:
I have a question, we have a couple of ways of stopping the node, one by killing it based on binary name (self.EXE), other by pid. Why do we need both? (I looked at your previous patch that added pid tracking but it wasn't entirely clear to me).

Reply:
This describes the original motivation https://github.com/redpanda-data/devprod/issues/883